extensions/common/crypto/ldp-verifiable-credentials/src/main/java/org/eclipse/edc/verifiablecredentials/linkeddata/LdpIssuer.java

```diff
@@ -0,0 +1,152 @@
+ *
+ */
+
+package org.eclipse.edc.verifiablecredentials.linkeddata;
+
+import com.apicatalog.jsonld.JsonLdReader;
+import com.apicatalog.jsonld.loader.DocumentLoader;
+import com.apicatalog.jsonld.loader.SchemeRouter;
+import com.apicatalog.ld.DocumentError;
+import com.apicatalog.ld.schema.LdProperty;
+import com.apicatalog.ld.signature.LinkedDataSignature;
+import com.apicatalog.ld.signature.SigningError;
+import com.apicatalog.ld.signature.key.KeyPair;
+import com.apicatalog.ld.signature.proof.EmbeddedProof;
+import com.apicatalog.ld.signature.proof.ProofOptions;
+import com.apicatalog.vc.VcTag;
+import com.apicatalog.vc.loader.StaticContextLoader;
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonValue;
+import org.eclipse.edc.jsonld.spi.JsonLd;
+import org.eclipse.edc.spi.result.Result;
+
+import java.net.URI;
+import java.util.Objects;
+
+public class LdpIssuer {
+    // mandatory properties
+    private JsonLd jsonLdService;
+    private DocumentLoader loader;
+    private boolean bundledContexts;
+    private URI base;
+
+    private LdpIssuer() {
+    }
+
+    public Result<JsonObject> signDocument(JsonObject document, KeyPair keyPair, ProofOptions proofOptions) {
+        Objects.requireNonNull(document, "Document must not be null");
+        Objects.requireNonNull(document, "KeyPair must not be null");
+        Objects.requireNonNull(document, "ProofOptions must not be null");
+        if (loader == null) {
+            // default loader
+            loader = SchemeRouter.defaultInstance();
+        }
+
+        if (bundledContexts) {
+            loader = new StaticContextLoader(loader);
+        }
+
+        return jsonLdService.expand(document)
+                .compose(expanded -> signExpanded(expanded, keyPair, proofOptions));
+
+    }
+
+    private Result<JsonObject> signExpanded(JsonObject expanded, KeyPair keyPair,
+                                            ProofOptions options) {
+
+        var optionalObject = JsonLdReader
+                .findFirstObject(expanded);
+        if (optionalObject.isEmpty()) {
+            return Result.failure("Error reading document: %s".formatted(DocumentError.ErrorType.Invalid));
+        }
+        var object = optionalObject.get();
+
+        if (options.getSuite() == null) {
+            return Result.failure("Unsupported Crypto Suite: %s".formatted(SigningError.Code.UnsupportedCryptoSuite));
+        }
+
+        var data = EmbeddedProof.removeProof(object);
+
+        var ldSignature = new LinkedDataSignature(options.getSuite().getCryptoSuite());
+
+        JsonObject proof;
+        try {
+            proof = options.getSuite().getSchema().write(options.toUnsignedProof());
+        } catch (DocumentError e) {
+            return Result.failure("Error writing proof: %s".formatted(e.getMessage()));
+        }
+
+        LdProperty<byte[]> proofValueProperty = options.getSuite().getSchema().tagged(VcTag.ProofValue.name());
```

Comment:
Can this be a var, I can't tell what the signature of tagged() is?

Reply:
`tagged()` is a parameterized method, so using `var` would make the `proofValueProperty` a `LdProperty<Object>`. This isn't a problem, because `proofValueProperty` is never used strongly typed as a method argument etc., so using `var` would be possible, although we'd lose the explicitness of `LdProperty<byte[]>`. I am good either way.

Reply:
OK, that's what I couldn't tell from the diff. I'm fine with the explicitness.